worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/TalonEmailSplitter.java

```diff
@@ -0,0 +1,313 @@
+        RE_ANDROID_WROTE,
+        RE_POLYMAIL
+    );
+
+    private final static int SPLITTER_MAX_LINES = 6;
+
+    private final static int MAX_LINES_COUNT = 1000;
+
+    private final static Pattern RE_LINK = Pattern.compile(
+        "<(http://[^>]*)>");
+
+    private final static Pattern QUOT_PATTERN = Pattern.compile(
+        "^>+ ?");
+
+    // Regular expression to identify if a line is a header.
+    private final static Pattern RE_HEADER = Pattern.compile(
+        ": ");
+
+    /**
+     * Given a message(which may consist of an email conversation thread with multiple emails), mark the lines to identify split lines,
+     * content lines and empty lines.
+     *
+     * Correct the split line markers inside header blocks. Header blocks are identified by the regular expression RE_HEADER.
+     *
+     * @param msgBody The email's message body
+     * @return The corrected markers
+     */
+    public static String splitEmails(final String msgBody)
+    {
+        // Given a message(which may consist of an email conversation thread with multiple emails), mark the lines to identify split
+        // lines content lines and empty lines.
+        //
+        // Correct the split line markers inside header blocks. Header blocks are identified by the regular expression RE_HEADER.
+        //
+        // Return the corrected markers.
+
+        final String msgBodyWithLinkBracketsReplaced = replaceLinkBrackets(msgBody);
+
+        // Don't process too long messages
+        final String[] linesArray = msgBodyWithLinkBracketsReplaced.split("\\r?\\n");
+        final String[] lines = (String[]) Arrays.copyOfRange(linesArray, 0, Math.min(linesArray.length, MAX_LINES_COUNT));
+
+        final String markers = removeInitialSpacesAndMarkMessageLines(lines);
+
+        final String markersWithQuotedEmailSplitlines = markQuotedEmailSplitlines(markers, lines);
+
+        // We don't want splitlines in header blocks
+        final String markersWithCorrectedSplitlinesInHeaders = correctSplitlinesInHeaders(markersWithQuotedEmailSplitlines, lines);
+
+        return markersWithCorrectedSplitlinesInHeaders;
+    }
+
+    private static String replaceLinkBrackets(final String msgBody)
+    {
+        // Normalize links i.e. replace '<', '>' wrapping the link with some symbols so that '>' closing the link couldn't be mistakenly
+        // taken for quotation marker.
+        //
+        // Converts msg_body into a unicode.
+
+        final StringBuffer stringBuffer = new StringBuffer();
+        final Matcher matcher = RE_LINK.matcher(msgBody);
+        while (matcher.find()) {
+            final int newlineIndex = msgBody.substring(0, matcher.start()).lastIndexOf("\n");
+            if (msgBody.charAt(newlineIndex + 1) == '>') {
+                matcher.appendReplacement(stringBuffer, matcher.group());
+            } else {
+                matcher.appendReplacement(stringBuffer, String.format("@@%s@@", matcher.group(1)));
```

Comment:
Just avoid the `String.format()` altogether.
```suggestion
                matcher.appendReplacement(stringBuffer, "@@" + matcher.group(1) + "@@");
```